cmd/libsnap-confine-private/bpf-support.c

```diff
@@ -161,3 +163,33 @@ int bpf_map_delete_elem(int map_fd, const void *key) {
 
     return sys_bpf(BPF_MAP_DELETE_ELEM, &attr, sizeof(attr));
 }
+
+#ifndef BPF_FS_MAGIC
+#define BPF_FS_MAGIC 0xcafe4a11
+#endif
+
+bool bpf_is_bpffs(const char *path) {
+    struct statfs fs;
+    int res = statfs(path, &fs);
+    if (res < 0) {
+        if (errno == ENOENT) {
+            /* no path at all */
+            return false;
+        }
+        die("cannot check filesystem type of %s", path);
+    }
+    if (fs.f_type == BPF_FS_MAGIC) {
+        return true;
+    }
+    return false;
```

Comment:
I'm not a fan of 1-liners, but in this case I'd write:
```suggestion
    return fs.f_type == BPF_FS_MAGIC;
```
Anyway, super nitpick.